File: 1. 两数之和/Solution.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        # 使用哈希表（这里是字典）
        n = len(nums)
        numbers = {}
        ans = []
        for i in range(n):
            if (target - nums[i]) in numbers.keys():
                ans.append(i)
                ans.append(numbers[target-nums[i]])
                break
            else:
                numbers[nums[i]] = i
        return ans






# class Solution:
#     def twoSum(self, nums: List[int], target: int) -> List[int]:
#         n = len(nums)
#         ans = []
#         for i in range(n):
#             sum = nums[i]
#             for j in range(n - 1 - i):
#                 sum += nums[i + 1 + j]
#                 if sum == target:
#                     ans.append(i)
#                     ans.append(i+1+j)
#                     return ans
#                 sum = nums[i]

File: 1. 两数之和/Solution.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] ans = new int[2];
        Map<Integer, Integer> map = new HashMap<>();
        for(int i = 0; i < nums.length; i++){
            if(map.containsKey(target - nums[i])){
                ans[0] = map.get(target-nums[i]);
                ans[1] = i;
                break;
            }
            else{
                map.put(nums[i], i);
            }
        }
        return ans;

    }
}





// class Solution {
//     public int[] twoSum(int[] nums, int target) {
//         int n=nums.length;
//         for(int i=0;i<n;i++){
//             for(int j=i+1;j<n;j++){
//                 if(nums[i]+nums[j]==target){
//                     return new int[]{i,j};
//                 }
//             }
//         }
//         return new int[]{0};

//     }
// }